module/VuFind/src/VuFind/Db/Service/UserListService.php

```diff
@@ -106,21 +109,24 @@ public function getUserListById(int $id): UserListEntityInterface
      */
     public function getPublicLists(array $includeFilter = [], array $excludeFilter = []): array
     {
-        $callback = function ($listOrId) {
-            return $listOrId instanceof UserListEntityInterface ? $listOrId->getId() : $listOrId;
-        };
-        $includeIds = array_map($callback, $includeFilter);
-        $excludeIds = array_map($callback, $excludeFilter);
-        $callback = function ($select) use ($includeIds, $excludeIds) {
-            $select->where->equalTo('public', 1);
-            if ($excludeIds) {
-                $select->where->notIn('id', $excludeIds);
-            }
-            if ($includeIds) {
-                $select->where->in('id', $includeIds);
-            }
-        };
-        return iterator_to_array($this->getDbTable('UserList')->select($callback));
+        $dql = 'SELECT ul FROM ' . UserListEntityInterface::class . ' ul ';
+
+        $parameters = [];
+        $where = ["ul.public = '1'"];
+        if (!empty($includeFilter)) {
+            $where[] = 'ul.id IN (:includeFilter)';
+            $parameters['includeFilter'] = $includeFilter;
+        }
+        if (!empty($excludeFilter)) {
+            $where[] = 'ul NOT IN (:excludeFilter)';
```

Comment:
Might have noticed a small error here, but not totally sure. Here the list is being compared and in include filter its comparing the id.

Reply:
Yes, this definitely merits deeper testing. I'll try to find time to take a closer look soon!

Reply:
I did some experimentation and found that the syntax is equivalent -- whether you use `ul` or `ul.id`, the query works correctly when you pass in an array of integers or an array of entity objects. But to make this more consistent, I've pushed a commit to use `ul` in both places so it's less confusing!